homeassistant/components/aemet/sensor.py

```diff
@@ -66,8 +66,6 @@ async def async_setup_entry(hass, config_entry, async_add_entities):
 class AbstractAemetSensor(CoordinatorEntity, SensorEntity):
     """Abstract class for an AEMET OpenData sensor."""
 
-    _attr_extra_state_attributes = {ATTR_ATTRIBUTION: ATTRIBUTION}
```

Comment:
Bit mixed feelings here. If this is the only extra state attributes used, why wouldn't we make it static ? That way we are not maintaining a dictionary per entity. Or maybe we wrap it in a `MappingProxy` so it raises when tried to update ?

Reply:
MappingProxyType could be ok. I don't think it's worth the risk to have mutable class attributes that are not meant to be shared.

Reply:
Just to clarify: Would it be enough to annotate `_attr_extra_state_attributes` with `Mapping` (in the `Entity` base class), i.e. to have `mypy` catch errors? Or should I change the dicts to `MappingProxyType`?

Although not completely perfect, adding `Mapping` could be enough IMO. That would also work perfectly for the remaining attributes I mentioned before by using `Sequence`.

Reply:
We'd need to make the dict read only by wrapping it in a MappingProxyType.